src/controllers/auth.js

```diff
@@ -0,0 +1,162 @@
+const bcrypt = require('bcrypt');
+const { Volunteer, Organization } = require('../models/user');
+
+exports.generalsignin = async (req, res) => {
+  const { email, password } = req.body;
+
+  const volunteer = await Volunteer.findOne({ email });
+  const organization = await Organization.findOne({ email });
+
+  const volunteerpassword = await Volunteer.findOne({ password });
+  const organizationpassword = await Organization.findOne({ password });
+
+  let volunteerororganization = null;
+
+  if (volunteer && volunteerpassword) {
+    volunteerororganization = volunteer;
+  } else if (organization && organizationpassword) {
+    volunteerororganization = organization;
+  }
+
+  if (!volunteerororganization) {
+    return res.status(400).json({ error: 'Wrong username,email or password' });
+  }
+
+  const isValidPassword = await bcrypt.compare(
+    password,
+    volunteerororganization.passwordhash
+  );
+  if (!isValidPassword) {
+    return res.status(400).json({ error: 'Wrong username,email or password' });
+  }
+  if (volunteer) {
+    req.user = volunteer;
+  } else if (organization) {
+    req.user = organization;
+  }
+  res.setHeader('user', volunteerororganization.id);
+  return res.status(200).json({
+    // eslint-disable-next-line no-underscore-dangle
+    organizationid: volunteerororganization._id,
+    message: `${req.user} signup is successfull.`,
+  });
+};
+
+exports.volunteersignup = async (req, res) => {
```

Comment:
To me, it does make more sense to implement signup in "user" routes but, I am not very confident in my opinion. I'll also ask Halit about this tomorrow.